src/editors/containers/VideoEditor/components/VideoSettingsModal/components/DurationWidget/hooks.js

```diff
@@ -0,0 +1,219 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { actions, selectors } from '../../../../../../data/redux';
+import messages from '../messages';
+
+import * as module from './hooks';
+
+const durationMatcher = /^(\d{0,2}):?(\d{0,2})?:?(\d{0,2})?$/i;
+
+export const durationWidget = ({ dispatch }) => {
+  const reduxStartStopTimes = useSelector(selectors.video.duration);
+  const setReduxStartStopTimes = (val) => dispatch(actions.video.updateField({ duration: val }));
+  const initialState = module.durationString(reduxStartStopTimes);
+  const [unsavedStartStopTimes, setUnsavedStartStopTimes] = useState(initialState);
+
+  useEffect(() => {
+    setUnsavedStartStopTimes(module.durationString(reduxStartStopTimes));
+  }, [reduxStartStopTimes]);
+
+  return {
+    reduxStartStopTimes,
+    unsavedStartStopTimes,
+    onBlur: (index) => (
+      (e) => module.updateDuration({
+        reduxStartStopTimes,
+        setReduxStartStopTimes,
+        unsavedStartStopTimes,
+        setUnsavedStartStopTimes,
+        index,
+        durationString: e.target.value,
+      })
+    ),
+    onChange: (index) => (
+      (e) => setUnsavedStartStopTimes(module.onDurationChange(unsavedStartStopTimes, index, e.target.value))
+    ),
+    onKeyDown: (index) => (
+      (e) => setUnsavedStartStopTimes(module.onDurationKeyDown(unsavedStartStopTimes, index, e))
+    ),
+    getTotalLabel: ({ duration, subtitle, intl }) => {
+      if (!duration.stopTime) {
+        if (!duration.startTime) {
+          return intl.formatMessage(messages.fullVideoLength);
+        }
+        if (subtitle) {
+          return intl.formatMessage(
+            messages.startsAt,
+            { startTime: module.durationStringFromValue(duration.startTime) },
+          );
+        }
+        return null;
+      }
+      const total = duration.stopTime - (duration.startTime || 0);
+      return intl.formatMessage(messages.total, { total: module.durationStringFromValue(total) });
+    },
+  };
+};
+
+/**
+ * durationString(duration)
+ * Returns the display value for embedded start and stop times
+ * @param {object} duration - object containing startTime and stopTime millisecond values
+ * @return {object} - start and stop time from incoming object mapped to duration strings.
+ */
+export const durationString = (duration) => ({
```

Comment:
This function, since it doesn't get hoisted, is better placed higher up in the file so it can be used without needing the "module.myFunction" call. Same for `durationStringFromValue`.